src/data/markdown/docs/01 guides/02 Using k6/14 Executors.md

```diff
@@ -0,0 +1,535 @@
+  over a 1 minute period:
+
+<div class="code-group" data-props='{"labels": [ "ramping-arr-rate.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    contacts: {
+      executor: 'ramping-arrival-rate',
+      startRate: 50,
+      timeUnit: '1s',
+      preAllocatedVUs: 50,
+      maxVUs: 100,
+      stages: [
+        { target: 200, duration: '30s' },
+        { target: 0, duration: '30s' },
+      ],
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/contacts.php');
+}
+```
+
+</div>
+
+
+### Externally controlled
+
+Control and scale execution at runtime via [k6's REST API](/misc/k6-rest-api) or
+the [CLI](https://k6.io/blog/how-to-control-a-live-k6-test).
+
+This executor formalizes what was previously possible globally, and using it is
+required in order to use the `pause`, `resume`, and `scale` CLI commands. Also note
+that arguments to `scale` to change the amount of active or maximum VUs only affect
+the externally controlled executor.
+
+| Option     | Type    | Description                                         | Default |
+|------------|---------|-----------------------------------------------------|---------|
+| `vus`      | integer | Number of VUs to run concurrently.                  | -       |
+| `maxVUs`   | integer | Maximum number of VUs to allow during the test run. | -       |
+| `duration` | string  | Total test duration.                                | -       |
+
+#### Examples
+
+- Execute a test run controllable at runtime, starting with 0 VUs up to a maximum of
+  50, and a total duration of 10 minutes:
+
+<div class="code-group" data-props='{"labels": [ "externally-controlled.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    contacts: {
+      executor: 'externally-controlled',
+      vus: 0,
+      maxVUs: 50,
+      duration: '10m',
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/contacts.php');
+}
+```
+
+</div>
+
+
+## Arrival rate
+
+TODO (expand on https://github.com/loadimpact/k6/issues/550)
+
+
+## Graceful stop and ramp down
+
+In versions before v0.27.0, k6 would interrupt any iterations being executed if the
+test is done or when ramping down VUs when using the `stages` option. This behavior
+could cause skewed metrics and wasn't user configurable.
+
+In v0.27.0 a new option is introduced for all executors (except externally-controlled):
+`gracefulStop`. With a default value of `30s`, it specifies the time k6 should wait
+for iterations to complete before forcefully interrupting them.
+
+A similar option exists for the ramping VUs executor: `gracefulRampDown`. This
+specifies the time k6 should wait for any iterations in progress to finish before
+VUs are returned to the global pool during the ramp down period defined in `stages`.
```

Comment:
I'd change "*the* ramp down period" to "*a* ramp down period" as ramp downs aren't mandatory and there can be more than one :)

Reply:
Resolved in 26911c0.